Console.Flashcards.mefdev/FlashcardApp.Console/Program.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Extensions.Configuration;
+using FlashcardApp.Console.IoC;
+using FlashcardApp.Console.Menus;
+using FlashcardApp.Console.MessageLoggers;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Program
+{
+    public static IConfiguration Configuration { get; set; }
+
+    public async static Task Main(string[] args)
+    {
+        try
+        {
+            LoadAppSettingConfiguration();
+            var serviceProvider = LoadIocConfiguration();
+            while (true)
+            {
+                await LoadDisplayMenu(serviceProvider);
+            }
+        }
+        catch (Exception ex)
+        {
+            MessageLogger.DisplayErrorMessage(ex.Message);
+        }
+    }
+
+    private static async Task LoadDisplayMenu(IServiceProvider serviceProvider)
+    {
+        var mainMenu = serviceProvider.GetRequiredService<MainMenu>();
+        await mainMenu.DisplayMenu();
+    }
+
+    private static IServiceProvider LoadIocConfiguration()
+    {
+        string connectionString = GetConnectionString();
+        var services = new ServiceCollection();
+        DependencyInjectionConfig.ConfigureServices(services, connectionString);
+        return services.BuildServiceProvider();
+    }
+
+    private static void LoadAppSettingConfiguration(){
+        var builder = new ConfigurationBuilder()
+            .SetBasePath(GetCurrentPath())
+            .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true);
```

Comment:
Note: your repo is missing/excluding appsettings.json, requiring a manual creation. I note that you document this in your README, however it is better to include this. If your appsettings has sensitive information, you can make an appsettings.development.json and get the project to read from this. You can then exclude this from source control.
Also, you can configure the following properties so you dont have to do replace logic that may break the app in future (i.e. a .NET upgrade)
![image](https://github.com/user-attachments/assets/7a67395d-184a-42ff-89f5-9711099af615)

Reply:
This post may also help:
https://learn.microsoft.com/en-us/answers/questions/1167351/how-to-create-environments-with-appsettings

Reply:
I've used appsettings.development.json now and included a dummy connection string to start with.